### PR TITLE
Bx update events

### PR DESCRIPTION
**Pull request by bxtsang**
Change logic to check.

1. Add recurr_id into query so that frontend will have recurr_id of existing events
2. Use recurr_id of existing event to check if it is Single or Recurring event currently
3. Use recurringData.frequency to check if it is being updated to Single or Recurring event

This should allow us to identify the 2 cases of:
1. single to recurring 
2. recurring to single

With this, frontend has no need to do any checks, just need to send the event object as it is and update the fields that the user inputs into the form.

### CODE REVIEW

File: backend/lambda/updateEventOrRecurring/lambda_function.py [hunk 1]
```diff
@@ -19,7 +19,7 @@ def lambda_handler(event, context):
         }
     
     # Single event --> Recurring event
-    if ('recurr_id' in eventsOrRecurring and not eventsOrRecurring['recurr_id']) and ("id" not in eventsOrRecurring['recurringData'] or not eventsOrRecurring['recurringData']['id']):
+    if eventsOrRecurring['recurr_id'] is None and eventsOrRecurring['recurringData']['frequency'] != 'None':
```
Comment (bxtsang) [inline backend/lambda/updateEventOrRecurring/lambda_function.py:22]: If recurr_id is None, it means currently the event is Single.

if the frequency sent over is not "None", means it is being updated into a recurring.

when these 2 are true, it is a case of Single to Recurring

File: backend/lambda/updateEventOrRecurring/lambda_function.py [hunk 2]
```diff
@@ -50,7 +50,7 @@ def lambda_handler(event, context):
         data = {"newEventData": eventsOrRecurring}
         
     # Recurring event --> Single event
-    elif ('recurr_id' in eventsOrRecurring and not eventsOrRecurring['recurr_id']) and "id" in eventsOrRecurring['recurringData']:
+    elif eventsOrRecurring['recurr_id'] is not None and eventsOrRecurring['recurringData']['frequency'] == 'None':
```
Comment (bxtsang) [inline backend/lambda/updateEventOrRecurring/lambda_function.py:53]: when recurr_id is not None, that means currently the event is recurring

when the frequency set by the user is "None", that means it is being set to single

when these 2 conditions are true, it is a case of Recurring to Single

**Review by bxtsang — COMMENTED** [review]
Add recurr_id so that frontend does not need to change this field

**Review by arixpsy — APPROVED** [review]
noted

**Review by glensee — APPROVED** [review]
Looks good

**Comment by Cliffenlee** [comment]
brilliant

**Pull request merged**